From 2ca3b6fd1cec57f332f26af62832d0fd9604e715 Mon Sep 17 00:00:00 2001
From: Alex Brett <alex.brett@citrix.com>
Date: Wed, 9 Dec 2020 10:31:47 +0000
Subject: [PATCH 5/8] Backport: CA-349929: Fix handling of /etc/centos-release
 for CentOS 8.3+

The format of /etc/centos-release has changed to not include any additional
content after the version number, which was not being matched by our existing
regular expression. Update the expression so it works with both the original
(e.g. CentOS 7 format), and the newer format.

Signed-off-by: Alex Brett <alex.brett@citrix.com>
---
 mk/xe-linux-distribution | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mk/xe-linux-distribution b/mk/xe-linux-distribution
index dc9fa4a..ed628a9 100755
--- a/mk/xe-linux-distribution
+++ b/mk/xe-linux-distribution
@@ -152,7 +152,7 @@ identify_redhat()
                -e 's/^Fedora.*release \([0-9]*\) (.*)$/distro=fedora;major=\1/gp;' \
                -e 's/^CentOS release \([0-9]*\)\.\([0-9]*\) (.*)/distro=centos;major=\1;minor=\2/gp;' \
                -e 's/^CentOS release \([0-9]*\) (.*)/distro=centos;major=\1/gp;' \
-               -e 's/^CentOS Linux release \([0-9]*\)\.\([0-9]*\)\(\.[0-9]*\)\? (.*)/distro=centos;major=\1;minor=\2/gp;' \
+               -e 's/^CentOS Linux release \([0-9]*\)\.\([0-9]*\).*$/distro=centos;major=\1;minor=\2/gp;' \
                -e 's/^CloudLinux\( Linux|Server\)\? release \([0-9]*\)\.\([0-9]*\)\(\.[0-9]*\)\? (.*)/distro=cloudlinux;major=\2;minor=\1/gp;' \
                -e 's/^CloudLinux release \([0-9]*\)\.\([0-9]*\)\(\.[0-9]*\)\? (.*)/distro=cloudlinux;major=\1;minor=\2/gp;' \
                -e 's/^Enterprise Linux Enterprise Linux .* release \([0-9]*\)\.\([0-9]*\) (.*)$/distro=oracle;major=\1;minor=\2;/gp;' \
-- 
2.30.2

